io-engine/src/grpc/v1/replica.rs

```diff
@@ -468,14 +560,18 @@ impl ReplicaRpc for ReplicaService {
                         };
                     // create snapshot
                     match lvol.create_snapshot(snap_config.clone()).await {
-                        Ok(()) => {
-                            info!("Create Snapshot Success for lvol: {lvol:?}");
+                        Ok(Some(snap_lvol)) => {
+                            info!("Create Snapshot Success for {lvol:?}, {snap_lvol:?}");
+                            let snapshot_descriptor =
+                                ReplicaSnapshotDescriptor::new(snap_lvol, lvol.uuid(), lvol.size());
                             Ok(CreateReplicaSnapshotResponse {
-                                snapshot_uuid: snap_config
-                                    .txn_id()
-                                    .unwrap_or_default(),
+                                replica_uuid: lvol.uuid(),
+                                snapshot: Some(ReplicaSnapshot::from(snapshot_descriptor)),
                             })
                         }
+                        Ok(None) => {
+                            todo!("Replica Snapshot will not hit this path");
```

Comment:
suggest changing from todo to something else, maybe a plain panic or assert

Reply:
It is removed now, as function split is done for local and remote and `Option` is removed now.